nltk/translate/chrf_score.py

```diff
@@ -176,6 +175,8 @@ def corpus_chrf(references, hypotheses, min_len=1, max_len=6, beta=3.0,
 
             n_gram_fscores[order].append(f_score)
 
+    num_sents = len(n_gram_fscores[min_len])
```

Comment:
+1 Much better, but there's this might be even better! `num_sents = len(hypotheses)`

Reply:
Your proposed solution won't work for generators. If you run `len(hypotheses)`, you'll exhaust the generator. This would break chrf3 usage in stream-based scenarios.

Reply:
Yeah, that's true. Hmmm, maybe using the original `count=0; count+=1` code is more readable and easier to maintain. Sorry for the confusion. 

Let me think about it again, there must be a better way =)

Reply:
One way would be to simply not care about the streaming case. Maybe I'm being too "defensive" here...

Reply:
I was being silly, `len(hypotheses)` is already used in the function. I'm using it to get `num_sents` as well. Let's not worry about generators for now?